contrib/99designs/gqlgen/option.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package gqlgen
+
+import (
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+const defaultServiceName = "gqlgen"
+
+// Instances of Option can be passed to New to configure the integration.
+type Option func(t *trace)
```

Comment:
The `Option` function takes a configuration structure, not the structure which it configures for. It's more common and better separation of concerns. Check out the other integrations.

I realise what you did seems "easier", but it arguably mixes concerns and becomes slightly inconsistent with our other integrations.

Reply:
I can just have the config be a member of the tracer